flent/metadata.py

```diff
@@ -514,3 +516,39 @@ def get_module_versions():
             module_versions[m] = v
 
     return module_versions
+
+def get_wifi_data(iface):
+    wifi_data = {}
+
+    unwanted_keys = ["Interface", "ifindex", "wdev", "wiphy"]
+
+    output = get_command_output("iw dev %s info" % iface)
+    if output is not None:
+
+        for line in output.splitlines():
+
+            if len(line.strip()) >= 1:
+                parts = line.split() 
+                k,v = parts[0], parts[1]
+
+                if k  in unwanted_keys:
+                  continue
+
+                if k == 'txpower':
+                   v = float(parts[1])
+
+                if k == 'channel':
+                    # This condition will return a dict with all the values of the channel
+		            # Output will be {'addr':..., channel': {'band': 2462, 'center1': 2462, 'number': 11, 'width': 20}, 'ssid':...}
```

Comment:
The indentation is off on the second comment line. Also, please add an example of the input (i.e., the line you're trying to parse) along with the output.